File: September/24/stairclimb.java
// LeetCode 70: Climbing Stairs
// Explanation:
// 1. Equivalent to Fibonacci sequence.
// 2. Use iteration with two variables.
// Time Complexity: O(n)
// Space Complexity: O(1)

class Solution {
    public int climbStairs(int n) {
        if (n <= 2) return n;
        int a = 1, b = 2;
        for (int i = 3; i <= n; i++) {
            int temp = a + b;
            a = b;
            b = temp;
        }
        return b;
    }
}

// Example usage:
// Solution sol = new Solution();
// System.out.println(sol.climbStairs(2)); // 2
// System.out.println(sol.climbStairs(3)); // 3

File: September/24/stairclimb.py
# LeetCode 70: Climbing Stairs
# Explanation:
# 1. This is Fibonacci: ways(n) = ways(n-1) + ways(n-2).
# 2. Use DP to compute result iteratively.
# Time Complexity: O(n)
# Space Complexity: O(n)

class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 2:
            return n
        dp = [0] * (n+1)
        dp[1], dp[2] = 1, 2
        for i in range(3, n+1):
            dp[i] = dp[i-1] + dp[i-2]
        return dp[n]
